mne/time_frequency/tfr.py

```diff
@@ -1238,9 +1264,15 @@ def tfr_multitaper(inst, freqs, n_cycles, time_bandwidth=4.0, use_fft=True,
     """
 
     data = _get_data(inst, return_itc)
-    picks = pick_types(inst.info, meg=True, eeg=True)
-    info = pick_info(inst.info, picks)
+    info = inst.info
+
+    if picks is None:
+        picks = pick_types(info, meg=True, eeg=True, ref_meg=False,
+                           exclude='bads')
+    picks = np.atleast_1d(picks)
```

Comment:
all your changes are the same lines. I would create a private function call

```
picks, info = _prepare_picks_info(picks, info)
```

for example